projects/build-env/src/plugin/targets/environment.targets.ts

```diff
@@ -0,0 +1,145 @@
+import type { ProjectConfiguration, TargetConfiguration } from '@nx/devkit';
+import type { NormalizedCreateNodeOptions } from '../normalize-create-nodes-options';
+import { join } from 'node:path';
+import { DEFAULT_NPM_INSTALL_TARGET } from './package.targets';
+import type { BuildEnvEnvironmentsOptions } from '../schema';
+import type { StartVerdaccioOptions } from '../../executors/bootstrap/verdaccio-registry';
+import { uniquePort } from '../../executors/bootstrap/unique-port';
+import { VERDACCIO_REGISTRY_JSON } from '../../executors/bootstrap/constants';
+
+export const DEFAULT_START_VERDACCIO_TARGET = 'build-env-verdaccio-start';
+export const DEFAULT_BOOTSTRAP_TARGET = 'build-env-env-bootstrap';
+export const DEFAULT_INSTALL_TARGET = 'build-env-env-install';
+export const DEFAULT_SETUP_TARGET = 'build-env-env-setup';
+export const DEFAULT_STOP_VERDACCIO_TARGET = 'build-env-verdaccio-stop';
+
+export function isEnvProject(
+  projectConfig: ProjectConfiguration,
+  options: NormalizedCreateNodeOptions['environments']
+): boolean {
+  const { tags: existingTags = [], targets } = projectConfig;
+  const existingTargetNames = Object.keys(targets ?? {});
+  const {
+    filterByTags: environmentsTagFilters,
+    targetNames: environmentTargetNames,
+  } = options;
+
+  if (!existingTargetNames || !environmentTargetNames) {
+    return false;
+  }
+
+  if (
+    existingTargetNames.some((existingTarget) =>
+      environmentTargetNames.includes(existingTarget)
+    )
+  ) {
+    if (existingTags && environmentsTagFilters) {
+      return existingTags.some((existingTag) =>
+        environmentsTagFilters.includes(existingTag)
+      );
+    }
+    return true;
+  }
+
+  return false;
+}
+
+export function verdaccioTargets(
+  projectConfig: ProjectConfiguration,
+  options: Pick<
+    NormalizedCreateNodeOptions['environments'],
+    'environmentsDir'
+  > &
+    Omit<StartVerdaccioOptions, 'projectName'>
+): Record<string, TargetConfiguration> {
+  const { name: envProject } = projectConfig;
+  const { environmentsDir, ...verdaccioOptions } = options;
+  const environmentDir = join(environmentsDir, envProject);
+
```

Comment:
would this make sense to check in this function if project is of type 'e2e'?

Reply:
I guess it should not be a strong requirement.